ci/scripts/towncrier_automation.py

```diff
@@ -0,0 +1,89 @@
+#!python3
+from __future__ import annotations
+
+import argparse
+import subprocess
+from typing import TYPE_CHECKING
+
+from packaging import version
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+
+class Args(argparse.Namespace):
+    version: str
+    dry_run: bool
+
+
+def parse_args(argv: Sequence[str] | None = None) -> Args:
+    parser = argparse.ArgumentParser(
+        prog="towncrier-automation",
+        description=(
+            "This script runs towncrier for a given version, "
+            "creates a branch off of the current one, "
+            "and then creates a PR into the original branch with the changes. "
+            "The PR will be backported to main if the current branch is not main."
+        ),
+    )
+    parser.add_argument("version", type=str, help="What the new version is")
```

Comment:
I changed this to a positional parameter. I’d generally do positional parameters for the single mandatory thing the command does. In this case releasing a version.

Before my change, if you didn’t specify `--version`, you’d get “TypeError: Expected str, not None” on the `version.parse(args.version)` line. Now you get “Missing mandatory parameter `version`”